src/query/expression/src/kernels/concat.rs

```diff
@@ -59,41 +59,55 @@ impl DataBlock {
         if blocks.is_empty() {
             return Err(ErrorCode::EmptyData("Can't concat empty blocks"));
         }
+        let block_refs = blocks.iter().collect::<Vec<_>>();
 
         if blocks.len() == 1 {
             return Ok(blocks[0].clone());
         }
 
-        let concat_columns = (0..blocks[0].num_columns())
-            .map(|i| {
-                debug_assert!(
-                    blocks
-                        .iter()
-                        .map(|block| &block.get_by_offset(i).data_type)
-                        .all_equal()
-                );
-
```

Comment:
We can check scalar value here like this, which is more clear.
```rust
let entry = blocks[0].get_by_offset(i);
if matches!(entry.value, Value::Scalar(_)) {
    if blocks
        .iter()
        .all(|b| b.get_by_offset(i) == entry)
    {
        return Ok(entry.clone());
    }
}
```

Reply:
add some unit test for this case is better.

Reply:
done, thank you